python/src/com/jetbrains/python/refactoring/move/moduleMembers/PyMoveSymbolProcessor.java

```diff
@@ -214,9 +215,19 @@ private static void insertQualifiedImportAndReplaceReference(@NotNull PsiNamedEl
     final LanguageLevel languageLevel = LanguageLevel.forElement(expression);
     if (srcFile != expression.getContainingFile()) {
       final QualifiedName qualifier = QualifiedNameFinder.findCanonicalImportPath(srcFile, expression);
-      PyPsiRefactoringUtil.insertImport(expression, srcFile, null, false);
-      final String newQualifiedReference = qualifier + "." + expression.getReferencedName();
-      expression.replace(generator.createExpressionFromText(languageLevel, newQualifiedReference));
+      if (PyCodeInsightSettings.getInstance().PREFER_FROM_IMPORT) {
+        // Respect preference: use from-import of the module and `module.Name`
+        PyPsiRefactoringUtil.insertImport(expression, srcFile, null, true);
+        final String moduleName = qualifier.getLastComponent();
+        final String newQualifiedReference = moduleName + "." + expression.getReferencedName();
+        expression.replace(generator.createExpressionFromText(languageLevel, newQualifiedReference));
+      }
+      else {
+        // Default behavior: import the module and use fully qualified reference
```

Comment:
"Respect preference" and "Default behavior" don't fit here. i suggest removing them or extracting each branch into a method

Reply:
Fixed